src/Components/Product.js

```diff
@@ -1,22 +1,50 @@
-import React, { useEffect } from 'react';
+import React, { useEffect, useState } from 'react';
 import { useDispatch, useSelector } from 'react-redux';
-import { FetchProduct } from '../redux/API/productfetch';
+import { fetchProduct } from '../redux/API/productfetch';
+import Item from './Item';
 
-const ProductLists = () => {
+const Products = () => {
   const dispatch = useDispatch();
-  const Products = useSelector((state) => state.Products);
+  const products = useSelector((state) => state.products);
   useEffect(() => {
-    if (!Products.length) {
-      dispatch(FetchProduct);
-    }
+    dispatch(fetchProduct());
   }, []);
-  return (
-
-    <div>
-      <div>Get list of Products</div>
 
-    </div>
+  const [searchTerm, setSearchTerm] = useState('');
+  const filteredproducts = products.filter((product) => product.title
+    .toLowerCase().includes(searchTerm.toLowerCase()));
+  const handleChange = (e) => {
+    const { value } = e.target;
+    setSearchTerm(value);
+  };
+  return (
+    <>
+      <div className="welcome">Welcome Home Page</div>
+      <input
+        type="text"
+        name="search"
+        className="search"
+        placeholder="Search Item"
+        value={searchTerm}
+        onChange={handleChange}
+      />
+      <div className="products">
+        {filteredproducts.map((product) => (
+          <Item
+            key={product.id}
+            id={product.id}
+            title={product.title}
+            price={product.price}
+            description={product.description}
+            category={product.category}
+            image={product.image}
+            rate={product.rating.rate}
+            count={product.rating.count}
+          />
+        ))}
+      </div>
+    </>
```

Comment:
- [ ] Talking about the design, I think there is still a lot of work to be done 😮 This web app is supposed to have a mobile-first responsive design and that is why you have been provided a mobile design to follow.
![image](https://user-images.githubusercontent.com/90222110/189451464-93951b1d-c0b9-4682-a754-f5b686c99c96.png)
However, your design is not looking appropriately on mobile screens so please make it responsive and make sure it looks very well on all mobile screens as it is focused on mobile 👍 

**Your Design**
![image](https://user-images.githubusercontent.com/90222110/189451315-4d8ce20b-cd1e-4507-82b7-cc127358a8b6.png)